src/EnergyPlus/api/autosizing.h

```diff
@@ -0,0 +1,121 @@
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef ENERGYPLUS_AUTOSIZING_H
+#define ENERGYPLUS_AUTOSIZING_H
+
+#include <EnergyPlus/api/TypeDefs.h>
+#include <EnergyPlus/api/EnergyPlusAPI.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// \file autosizing.h
+/// \brief This "Auto-sizing" API category provides access to auto-sizing calculations inside EnergyPlus.
+/// \details Auto-sizing is an important aspect of an EnergyPlus simulation, allowing users to gather simulation results
+///          for cases where the actual size of components and systems are not known ahead of time.
+
+/// \brief This typedef is a convenient pointer to an internal sizer class.
+/// \details Just one is declared here as any sizer object can fit in it.  An instance can be found by calling any
+///          sizer***New() function
+ENERGYPLUSLIB_API typedef void * Sizer;
+
+/// \brief Returns a new reference to a HeatingAirflowUA Sizer class
+/// \details The sizer class allows access to sizer stuff.
+/// \returns This functions allocates a new sizer class and returns a Sizer, which is a pointer to that new instance.
+/// \remark In API applications, when the calling script is done with the sizer instance, call `sizerHeatingAirflowUADelete` to clean up the instance.
+/// \see Sizer
+/// \see sizerHeatingAirflowUADelete
+ENERGYPLUSLIB_API Sizer sizerHeatingAirflowUANew();
+/// \brief Deletes an instance of a HeatingAirflowUA Sizer class
+/// \details When an instance of a Sizer class is created using `sizerHeatingAirflowUANew`, it should be cleaned up when totally done with it.
+/// \param[in] sizer An instance of a Sizer class to be deleted.  The Sizer class is initially created by calling `sizerHeatingAirflowUANew`.
+/// \see Sizer
+/// \see sizerHeatingAirflowUANew
+ENERGYPLUSLIB_API void sizerHeatingAirflowUADelete(Sizer sizer);
+
+// TODO: These need work.  Need to decide if they should be more/less specific, and we need to test all of them
+// TODO: Also, some of the code paths appear to want a mass flow rate while others want a volume flow rate, need to verify that
+
+/// \brief Initializes the HeatingAirflowUA sizer class
+/// \details All required data for setting up the HeatingAirflowUA sizer is passed in as arguments to this function.
+/// \param[in] sizer An instance of a Sizer class, which can be created by calling `sizerHeatingAirflowUANew`.
+/// \param[in] elevation The elevation above sea level for evaluating fluid properties, in m
+/// \remark This must be called prior to each call to sizerHeatingAirflowUACalculate
+/// \see Sizer
+/// \see sizerHeatingAirflowUACalculate
+/// \see sizerHeatingAirflowUADelete
+
+ENERGYPLUSLIB_API void sizerHeatingAirflowUAInitializeForSingleDuctZoneTerminal(Sizer sizer, Real64 elevation, Real64 mainFlowRate);
```

Comment:
Once we have settled on the names and configurations for all the possible initialization functions, I'll thoroughly document them here.